modules/tensor_mechanics/src/materials/ADTransverselyIsotropicCreepStressUpdate.C

```diff
@@ -0,0 +1,247 @@
+  params.addParam<Real>("gas_constant", 8.3143, "Universal gas constant");
+  params.addParam<Real>("start_time", 0.0, "Start time (if not zero)");
+  params.addRequiredParam<std::vector<Real>>("hill_constants",
+                                             "Hill material constants in order: F, "
+                                             "G, H, L, M, N");
+
+  return params;
+}
+
+ADTransverselyIsotropicCreepStressUpdate::ADTransverselyIsotropicCreepStressUpdate(
+    const InputParameters & parameters)
+  : ADAnisotropicReturnCreepStressUpdateBase(parameters),
+    _has_temp(isParamValid("temperature")),
+    _temperature(_has_temp ? coupledValue("temperature") : _zero),
+    _coefficient(getParam<Real>("coefficient")),
+    _n_exponent(getParam<Real>("n_exponent")),
+    _m_exponent(getParam<Real>("m_exponent")),
+    _activation_energy(getParam<Real>("activation_energy")),
+    _gas_constant(getParam<Real>("gas_constant")),
+    _start_time(getParam<Real>("start_time")),
+    _exponential(1.0),
+    _exp_time(1.0),
+    _hill_constants(6),
+    _qsigma(0.0)
+{
+  if (_start_time < _app.getStartTime() && (std::trunc(_m_exponent) != _m_exponent))
+    paramError("start_time",
+               "Start time must be equal to or greater than the Executioner start_time if a "
+               "non-integer m_exponent is used");
+
+  _hill_constants = getParam<std::vector<Real>>("hill_constants");
+
+  // Hill constants, some constraints apply
+  const Real F = _hill_constants[0];
+  const Real G = _hill_constants[1];
+  const Real H = _hill_constants[2];
+  const Real L = _hill_constants[3];
+  const Real M = _hill_constants[4];
+  const Real N = _hill_constants[5];
+
+  // Build Hill tensor or anisotropy matrix
+  ADDenseMatrix hill_tensor(6, 6);
+  hill_tensor(0, 0) = G + H;
+  hill_tensor(1, 1) = F + H;
+  hill_tensor(2, 2) = F + G;
+  hill_tensor(0, 1) = hill_tensor(1, 0) = -H;
+  hill_tensor(0, 2) = hill_tensor(2, 0) = -G;
+  hill_tensor(1, 2) = hill_tensor(2, 1) = -F;
+
+  hill_tensor(3, 3) = 2.0 * N;
+  hill_tensor(4, 4) = 2.0 * L;
+  hill_tensor(5, 5) = 2.0 * M;
+}
+
+void
+ADTransverselyIsotropicCreepStressUpdate::computeStressInitialize(
+    const ADDenseVector & /*effective_trial_stress*/,
+    const ADRankFourTensor & /*elasticity_tensor*/)
+{
+  if (_has_temp)
+    _exponential = std::exp(-_activation_energy / (_gas_constant * _temperature[_qp]));
+
+  _exp_time = std::pow(_t - _start_time, _m_exponent);
+}
+
+ADReal
+ADTransverselyIsotropicCreepStressUpdate::computeResidual(
+    const ADDenseVector & /*effective_trial_stress*/,
+    const ADDenseVector & stress_new,
+    const ADReal & delta_gamma)
+{
+  // Hill constants, some constraints apply
+  const Real F = _hill_constants[0];
+  const Real G = _hill_constants[1];
+  const Real H = _hill_constants[2];
+  const Real L = _hill_constants[3];
+  const Real M = _hill_constants[4];
+  const Real N = _hill_constants[5];
```

Comment:
Semantically it seems better to use `const &` here as you are declaring an alias. That way the compiler can substitute the RHS in below without ever crating a copy. If you don't use a reference there is no guarantee that LHS and RHS are the same (`_hill_constants` could be changed as a side-effect, especially since it is not `const` itself). Way too long of a comment for such a minor point...